cas_implementations/src/path.rs

```diff
@@ -13,8 +13,11 @@ pub fn storage_path(path: &Path, folder_name: &str) -> HcResult<PathBuf> {
 pub fn create_path_if_not_exists(path: &Path) -> HcResult<()> {
     if !path.exists() {
         return DirBuilder::new()
+            .recursive(true) // create parent dirs if necessary
             .create(path)
-            .map_err(|_| HolochainError::IoError("Could not create directory".to_string()));
+            .map_err(|_| {
+                HolochainError::IoError(format!("Could not create directory: {:?}", path))
+            });
```

Comment:
Okay, this is ridiculous... this whole function already exists in the `stdlib`. It's called `std::fs::create_dir_all`

Reply:
Hey @sphinxc0re, you definitely have a deeper knowledge of the stdlib than all of, so thanks for that.  It would be more helpful if you could just provide a suggested edit using github's UI, or even just check out the branch and make the fix.  Thanks.

Reply:
Ya, I saw that `std:fs::create_dir_all`, but didn't look into it too deeply -- I didn't want to get into changing crates, etc.  I'll take a look at it.  Thanks! ...  Ah; it takes a Path, not an &Path:
```
pub fn create_dir_all<P: AsRef<Path>>(path: P) -> io::Result<()> {
    DirBuilder::new().recursive(true).create(path.as_ref())
}
```
so we'd have to `.to_owned()` the path, and then call it.  I don't know if its worth it to make the implementation of the function more opaque.  Its just one line of method calls, and I think it can stay like it is...

Reply:
I made the change, copying the `path` (no biggie, we only use this once right now), but I don't think it improves the code.  I like it explicit and clear, the way it is. And, no copy.  Changing the API to a plain `Path` also was not better, because when you want to create a path, usually you want to do something with it later, requiring the (now moved) Path.  And, it launched a bunch of complaints by the compiler that you really should borrow the `Path`, and `[u8]` arrays and sizes and...  I don't want to waste time on investigating that right now.  Basically, the `std::fs::create_dirs_all` API is worse! :)